<launch>

  <arg name="rectified_scan_points_topic" default="/points_raw" description="Input rectified point cloud"/>
  <arg name="optimization_start_pose_topic" default="/optimization_start_pose" />
  <arg name="estimated_pose_topic" default="/estimated_pose" />
  <arg name="estimated_pose_with_covariance_topic" default="/estimated_pose_with_covariance" />

  <arg name="node_name" default="lidar_feature_localization" />

  <node pkg="lidar_feature_localization" exec="lidar_feature_localization" name="$(var node_name)" output="log">
    <remap from="points_raw" to="$(var rectified_scan_points_topic)"/>
    <remap from="optimization_start_pose" to="$(var optimization_start_pose_topic)" />
    <remap from="estimated_pose" to="$(var estimated_pose_topic)" />
    <remap from="estimated_pose_with_covariance" to="$(var estimated_pose_with_covariance_topic)" />
    <param from="$(find-pkg-share lidar_feature_localization)/config/extraction.param.yaml" />
  </node>
</launch>
